runtime/v2/shim.go

```diff
@@ -256,14 +256,35 @@ func makeConnection(ctx context.Context, address string, onClose func()) (_ io.C
 			}
 		}()
 
-		// TODO: figure out how to invoke onCloseWithShimLog callback when shim connection is closed.
-
-		return conn, nil
+		return &grpcConnUtil{
+			onCloseFn:  onClose,
+			ClientConn: conn,
+		}, nil
 	default:
 		return nil, fmt.Errorf("unexpected protocol: %q", params.Protocol)
 	}
 }
 
+// grpcConnUtil exists solely so we can have something similar to
+// ttrpc.WithOnClose to have a callback run when we Close the
+// connection. We could spawn a goroutine to monitor the connection
+// status until it changes to shutdown, but we'd have an extra
+// routine per shim in that case.
+type grpcConnUtil struct {
```

Comment:
Not a fan of this naming.. `grpcConnWithCallback` maybe? My end of work day brain can't think of something great

Reply:
just `grpcConn` maybe?

Reply:
That works for me